src/mpid/common/thread/mpidu_thread_fallback.h

```diff
@@ -295,7 +295,7 @@ M*/
             break;                                                      \
         MPIDU_Thread_mutex_unlock(mutex_ptr_, err_ptr_);                \
         MPIR_Assert(*err_ptr_ == 0);                                    \
-        MPL_thread_yield();                                             \
+        MPL_thread_yield(&(mutex_ptr_)->mutex);                                             \
```

Comment:
`MPL_thread_yield` is oblivious of mutexes by design. What's the reason for changing it? Can it be simply done at the MPIDU level?

Reply:
Hmm, I think what we need is adding a special version of `MPIDU_Thread_yield` that do not do lock and unlock and pass the mutex to `MPL_thread_yield`. The `MPL_thread_yield` internally choose `ABT_thread_yield` or lock/unlock+`MPL_sched_yield`. I can write that version up.

Reply:
That branch is related to `ult_yield_only`, which itself is related to MPIX_THREAD_ULT, which we agreed to drop. Basically any code related to these two key words should be removed